AirCasting/Utils/Connection/BluetoothConnection.swift

```diff
@@ -0,0 +1,29 @@
+import SwiftUI
+import CoreData
+import Resolver
+
+final class BluetoothConnectionProtector: ConnectionProtectable {
+    
+    enum BluetoothConnectionProtectorError: Error {
+        case alreadyConnected
+        case readError(Error)
+    }
+    
+    @Injected private var database: SessionsFetchable
+    
+    func isAirBeamAvailableForNewConnection(completion: @escaping (Result<Void, Error>) -> Void) {
```

Comment:
Are we really checking if AirBeam is available here? I think the name is misleading cause we are only checking if there is an active Airbeam session

Reply:
@anna1901 I think this name is accurate. On the implementation level indeed we're just checking for existing database AB session. But we're operating on a higher abstraction level here. We don't want clients of this protocol to be aware of any of this, they just want to check if they can connect to AB, they don't care how it's checked behind the scenes.

Imagine a hypothetical scenario where we're asked to disable connecting to older AB models. With this protocol and this function name we'd be able to just add another class that conforms to `ConnectionProtectable`, combine it with this one inside resolver registration (using a [composite pattern](https://refactoring.guru/design-patterns/composite)) and that's it, no existing code would be modified, very clean and extensible.
Now if we named this function `checkForExistingABSessions` we'd lose all of that flexibility. Just because of the name 🤯

Reply:
I came up with sth, we will see what you will say 😅